crypto/des/fcrypt.c

```diff
@@ -70,9 +70,12 @@ char *DES_crypt(const char *buf, const char *salt)
     char e_buf[32 + 1];         /* replace 32 by 8 ? */
     char *ret;
 
+    if (sizeof(salt) < 2) return NULL;
```

Comment:
Ups, I were testing sizeof against char salt[3]; and then it had sizeof(salt) == 3. My bad.
I wonder, commonly there are no NULL tests againts ptrs, in this one better write if (salt == NULL) return NULL; or just OpenSSLAssert()?